src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/PackageRestore/Snapshots/RestoreComparer.ReferenceItemEqualityComparer.cs

```diff
@@ -14,15 +14,31 @@ public override bool Equals(ReferenceItem? x, ReferenceItem? y)
                 if (!StringComparers.ItemNames.Equals(x.Name, y.Name))
                     return false;
 
-                return x.Properties.SequenceEqual(y.Properties, ReferenceProperties);
+                if (x.Properties.Count != y.Properties.Count)
+                    return false;
+
+                foreach (KeyValuePair<string, string> property in x.Properties)
+                {
+                    if (!y.Properties.TryGetValue(property.Key, out string yValue))
+                    {
+                        return false;
+                    }
+
+                    if (!StringComparer.Ordinal.Equals(Equals(property.Key, yValue)))
```

Comment:
Bug? Comparing key and value?

I think we have a semantic comparer for these. `StringComparers.PropertyValue` from memory.